src/PointwiseFunctions/Hydro/EquationsOfState/PiecewisePolytropicFluid.cpp

```diff
@@ -0,0 +1,281 @@
+          pow(transition_density_, polytropic_exponent_lo - 1.0)),
+      polytropic_constant_lo_(polytropic_constant_lo),
+      polytropic_exponent_lo_(polytropic_exponent_lo),
+      polytropic_constant_hi_(
+          polytropic_constant_lo *
+          pow(transition_density,
+              polytropic_exponent_lo - polytropic_exponent_hi)),
+      polytropic_exponent_hi_(polytropic_exponent_hi) {}
+
+EQUATION_OF_STATE_MEMBER_DEFINITIONS(template <bool IsRelativistic>,
+                                     PiecewisePolytropicFluid<IsRelativistic>,
+                                     double, 1)
+EQUATION_OF_STATE_MEMBER_DEFINITIONS(template <bool IsRelativistic>,
+                                     PiecewisePolytropicFluid<IsRelativistic>,
+                                     DataVector, 1)
+
+template <bool IsRelativistic>
+bool PiecewisePolytropicFluid<IsRelativistic>::operator==(
+    const PiecewisePolytropicFluid<IsRelativistic>& rhs) const {
+  return (transition_density_ == rhs.transition_density_) and
+         (polytropic_constant_lo_ == rhs.polytropic_constant_lo_) and
+         (polytropic_exponent_lo_ == rhs.polytropic_exponent_lo_) and
+         (polytropic_constant_hi_ == rhs.polytropic_constant_hi_) and
+         (polytropic_exponent_hi_ == rhs.polytropic_exponent_hi_);
+}
+
+template <bool IsRelativistic>
+bool PiecewisePolytropicFluid<IsRelativistic>::operator!=(
+    const PiecewisePolytropicFluid<IsRelativistic>& rhs) const {
+  return not(*this == rhs);
+}
+
+template <bool IsRelativistic>
+bool PiecewisePolytropicFluid<IsRelativistic>::is_equal(
+    const EquationOfState<IsRelativistic, 1>& rhs) const {
+  const auto& derived_ptr =
+      dynamic_cast<const PiecewisePolytropicFluid<IsRelativistic>* const>(&rhs);
+  return derived_ptr != nullptr and *derived_ptr == *this;
+}
+
+template <bool IsRelativistic>
+std::unique_ptr<EquationOfState<IsRelativistic, 1>>
+PiecewisePolytropicFluid<IsRelativistic>::get_clone() const {
+  auto clone =
+      std::make_unique<PiecewisePolytropicFluid<IsRelativistic>>(*this);
+  return std::unique_ptr<EquationOfState<IsRelativistic, 1>>(std::move(clone));
+}
+
+template <bool IsRelativistic>
+PiecewisePolytropicFluid<IsRelativistic>::PiecewisePolytropicFluid(
+    CkMigrateMessage* msg)
+    : EquationOfState<IsRelativistic, 1>(msg) {}
+
+template <bool IsRelativistic>
+void PiecewisePolytropicFluid<IsRelativistic>::pup(PUP::er& p) {
+  EquationOfState<IsRelativistic, 1>::pup(p);
+  p | transition_density_;
+  p | transition_pressure_;
+  p | transition_spec_eint_;
+  p | polytropic_constant_lo_;
+  p | polytropic_exponent_lo_;
+  p | polytropic_constant_hi_;
+  p | polytropic_exponent_hi_;
+}
+
+template <bool IsRelativistic>
+template <class DataType>
+Scalar<DataType>
+PiecewisePolytropicFluid<IsRelativistic>::pressure_from_density_impl(
+    const Scalar<DataType>& rest_mass_density) const {
+  auto polytropic_constant =
```

Comment:
All of these `make_with_value` do a memory allocation in the DataVector case, so I'm worried this will be quite slow. I would instead do
```cpp
for (size_t i = ;0 < get_size(get(rest_mass_density)); ++i) {
  const double density = get_element(get(rest_mass_density), i);
  if (density < transition) {
    get_element(result, i) = ...;
  } else {
   ...
  }
}
```
The `get_size` and `get_element` are defined in `Utilities/ContainerHelpers.hpp` and are specifically designed for the sort of case where we want to "loop" over a `double` or `DataVector` to avoid an allocation (or series of them). I think you can keep the `choose_polytropic_properties` function with some tweaks

We'll need to revisit completely when adding vectorization, but that's a future problem....